### PR TITLE
Add ConsoleKey.None and ConsoleModifiers.None

### LINKED ISSUE #79868: [API Proposal]: Add ConsoleModifiers.None
### Background and motivation

`ConsoleModifiers` is a flags enum. Per our guidelines, they should have a value for `0`, which is usually called `None`.

Since the value is missing, developers need to compare the enum to the value zero. This doesn't require a cast because in C# the literal `0` can be implicitly converted into any enum value, but this isn't intuitive. It also means when expecting the value in the debugger, one sees the value `0` instead of a more friendly text `None`.

### API Proposal

```C#
namespace System;

[Flags]
public enum ConsoleModifiers
{
    None = 0,

    // Existing:
    //
    // Alt = 1,
    // Shift = 2,
    // Control = 4
}
```

### API Usage

```C#
private void HandleKey(ConsoleKeyInfo info)
{
    switch (info.Modifiers, info.Key)
    {
        case (ConsoleModifiers.None, ConsoleKey.Q):
        case (ConsoleModifiers.None, ConsoleKey.Escape):
            Application.Quit();
            break;
        case (ConsoleModifiers.None, ConsoleKey.N):
            GotoNextSearchHit();
            break;
        case (ConsoleModifiers.Shift, ConsoleKey.N):
            GotoPreviousSearchHit();
            break;
    }    
}
```

### Alternative Designs

_No response_

### Risks

_No response_

### PR DESCRIPTION
**Pull request by terrajobst**
Submitted as draft PR to have a CI pass with tests. Will mark as ready for review when it passes.

This fixes #79868.

### CODE REVIEW

**Comment by terrajobst** [comment]
Unless I'm mistaken it doesn't seem the failures are related to this change, so I'm going to mark it ready.

File: src/libraries/System.Console/src/System/ConsoleKey.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@ namespace System
 {
     public enum ConsoleKey
     {
+        None = 0x0,
```
Comment (stephentoub) [inline src/libraries/System.Console/src/System/ConsoleKey.cs:8]: @carlossanlop, do these need XML docs?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Console/src/System/ConsoleKey.cs (hunk 1 above)
Comment (carlossanlop) [reply src/libraries/System.Console/src/System/ConsoleKey.cs:8]: Area owners can document these new enum values in dotnet-api-docs directly.

Assuming this change makes it into Preview3, that build will get released on April 11: https://dev.azure.com/devdiv/DevDiv/_wiki/wikis/DevDiv.wiki/31692/NET-8-Schedule . On that day, we will update dotnet-api-docs to start showing Preview3 in the MS Docs .NET version dropdown. When that happens, area owners can document the new enums in the dotnet-api-docs xmls:

- https://github.com/dotnet/dotnet-api-docs/blob/6a16408631322b76de318358be710f8dfbbbb5c1/xml/System/ConsoleKey.xml
- https://github.com/dotnet/dotnet-api-docs/blob/6a16408631322b76de318358be710f8dfbbbb5c1/xml/System/ConsoleModifiers.xml

By the way, I don't know if you also wanted to know this, but fortunately all the old enum values area already documented in dotnet-api-docs:

- https://learn.microsoft.com/en-us/dotnet/api/system.consolekey?view=net-7.0
- https://learn.microsoft.com/en-us/dotnet/api/system.consolemodifiers?view=net-7.0

So only the new two enum values will need docs.

Comment (terrajobst) [reply src/libraries/System.Console/src/System/ConsoleKey.cs:8]: So the process is we merge this first, tooling will update the XML files and then I can submit a PR to add the text?

Comment (stephentoub) [reply src/libraries/System.Console/src/System/ConsoleKey.cs:8]: The typical process is we use xml comments on new apis and then those are used to seed the docs. It sounds like Carlos is saying you could also wait and submit a pr to the docs repo for these later.

Comment (terrajobst) [reply src/libraries/System.Console/src/System/ConsoleKey.cs:8]: Sounds good. I also submitted a PR for that; if that's the wrong way of doing it, I can also wait until the tooling populated the files for the new members.

**Pull request merged**